pkg/alertmanager/operator.go

```diff
@@ -654,6 +657,17 @@ func (c *Operator) sync(ctx context.Context, key string) error {
 
 	level.Info(logger).Log("msg", "sync alertmanager")
 
+	// Check the existence of the storage class if not empty and if the operator has enough permissions.
+	if storageClassName := operator.StringPtrValOrDefault(am.Spec.Storage.VolumeClaimTemplate.Spec.StorageClassName, ""); c.canReadStorageClass && storageClassName != "" {
```

Comment:
sorry about suggesting the bug in the first place :)
```suggestion
	if am.Spec.Storage != nil {
		if storageClassName := operator.StringPtrValOrDefault(am.Spec.Storage.VolumeClaimTemplate.Spec.StorageClassName, ""); c.canReadStorageClass && storageClassName != "" {
```

Reply:
👋 -- We recently merged https://github.com/prometheus-operator/prometheus-operator/pull/5760, which adds the option to declare multiple VolumeClaims. Could we also check for storage classes inside this object too?